src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_x509.c

```diff
@@ -153,10 +154,52 @@ int32_t AndroidCryptoNative_X509DecodeCollection(const uint8_t *buf, int32_t buf
     return ret;
 }
 
-PAL_X509ContentType AndroidCryptoNative_X509GetContentType(const uint8_t *buf, int32_t len)
+int32_t AndroidCryptoNative_X509ExportPkcs7(jobject* /*X509Certificate[]*/ certs,
+                                            int32_t certsLen,
+                                            uint8_t* out,
+                                            int32_t* outLen)
+{
+    assert(certs != NULL && certsLen > 0);
+    assert(outLen != NULL);
+    JNIEnv* env = GetJNIEnv();
+
+    int32_t ret = FAIL;
+    INIT_LOCALS(loc, certList, certType, certFactory, certPath, pkcs7Type, encoded)
+
+    // ArrayList<Certificate> certList = new ArrayList<Certificate>();
+    // foreach (Certificate cert in certs)
+    //     certList.add(cert);
+    loc[certList] = (*env)->NewObject(env, g_ArrayListClass, g_ArrayListCtor, certsLen);
+    for (int i = 0; i < certsLen; ++i)
+    {
+        (*env)->CallBooleanMethod(env, loc[certList], g_ArrayListAdd, certs[i]);
+        ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+    }
+
+    // CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
+    loc[certType] = JSTRING("X.509");
+    loc[certFactory] = (*env)->CallStaticObjectMethod(env, g_CertFactoryClass, g_CertFactoryGetInstance, loc[certType]);
+    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+
+    // CertPath certPath = certFactory.generateCertPath(certList);
+    // byte[] encoded = certPath.getEncoded("PKCS7");
+    loc[certPath] = (*env)->CallObjectMethod(env, loc[certFactory], g_CertFactoryGenerateCertPathFromList, loc[certList]);
```

Comment:
Just confirming: This works for unrelated certificates?  It hasn't actually tried building the path yet?

Reply:
Going through `CertificateFactory` (this) works - it doesn't try building the path yet and isn't validated unless put through a `CertPathValidator`. Going through `CertPathBuilder` would not work, as it does do the path building.